src/main/java/org/opentripplanner/middleware/bugsnag/BugsnagDispatcher.java

```diff
@@ -0,0 +1,217 @@
+ *
+ * A bugsnag API key is a key that is unique to an individual Bugsnag user. This key can be obtained by logging into
+ * Bugsnag (https://app.bugsnag.com), clicking on settings (top right hand corner) -> “My account settings”. From here
+ * select “Personal auth tokens” and then “Generate new token”.
+ *
+ * The following Bugsnag API endpoints are currently used:
+ *
+ * - https://api.bugsnag.com/organizations/<organization_id>/event_data_requests
+ * - https://api.bugsnag.com/organizations/<organization_id>/event_data_requests/<event_data_request_id>
+ * - https://api.bugsnag.com/organizations/<organization_id>/projects
+ */
+public class BugsnagDispatcher {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BugsnagDispatcher.class);
+
+    private static final String BUGSNAG_API_URL = "https://api.bugsnag.com";
+    private static final String BUGSNAG_API_KEY = getConfigPropertyAsText("BUGSNAG_API_KEY");
+    private static final String BUGSNAG_ORGANIZATION = getConfigPropertyAsText("BUGSNAG_ORGANIZATION");
+    private static final int BUGSNAG_REPORTING_WINDOW_IN_DAYS
+        = getConfigPropertyAsInt("BUGSNAG_REPORTING_WINDOW_IN_DAYS", 14);
+
+    /** Organization from which all projects and event request information will be extracted. Set once during
+     * startup. */
+    private static Organization ORGANIZATION = null;
+
+    /** Headers that are required by Bugsnag for each request. */
+    private static HashMap<String, String> BUGSNAG_HEADERS = new HashMap<>();
+
+    /** Filter object defining the boundaries on which event requests will be based. */
+    private static ObjectNode eventRequestFilter;
+
+    private static int CONNECTION_TIMEOUT_IN_SECONDS = 5;
+
+    static {
+        setBugsnagRequestHeaders();
+        buildEventRequestFilter();
+        setOrganization();
+    }
+
+    /**
+     * Define the mandatory Bugsnag request headers.
+     */
+    private static void setBugsnagRequestHeaders() {
+        BUGSNAG_HEADERS.put("Authorization", "token " + BUGSNAG_API_KEY);
+        BUGSNAG_HEADERS.put("Accept", "application/json; version=2");
+        BUGSNAG_HEADERS.put("Content-Type", "application/json");
+    }
+
+    /**
+     * Build the event request filter that will be passed to Bugsnag with every event data request.
+     */
+    private static void buildEventRequestFilter() {
+        final ObjectMapper mapper = new ObjectMapper();
+        ObjectNode reportingWindowCondition = mapper.createObjectNode();
+        reportingWindowCondition.put("type", "eq");
+        reportingWindowCondition.put("value", BUGSNAG_REPORTING_WINDOW_IN_DAYS + "d");
+
+        ArrayNode sinceFilters = mapper.createArrayNode();
+        sinceFilters.add(reportingWindowCondition);
+
+        ObjectNode filters = mapper.createObjectNode();
+        filters.set("event.since", sinceFilters);
+
+        eventRequestFilter = mapper.createObjectNode();
+        eventRequestFilter.set("filters", filters);
```

Comment:
All of these filters need comments. Otherwise, it's not clear what they're doing.

Reply:
Updated.